### PR TITLE
add HIL_STATE_QUATERNION message

### PR DESCRIPTION
**Pull request by kd0aij**
provides simulated "ground truth" data for vehicle attitude, rotation rates and velocity 

requires a modification to jMAVlib also: https://github.com/PX4/jMAVlib/pull/1

### CODE REVIEW

File: src/me/drton/jmavsim/Simulator.java [hunk 1]
```diff
@@ -306,7 +306,7 @@ private AbstractMulticopter buildMulticopter() {
         I.m11 = 0.005;  // Y
         I.m22 = 0.009;  // Z
         vehicle.setMomentOfInertia(I);
-        vehicle.setMass(1.0); // mass of the vehicle, too light the vehicle will start climbing on arm
+        vehicle.setMass(0.8);
```
Comment (LorenzMeier) [inline src/me/drton/jmavsim/Simulator.java]: Can you explain the reason for this change? All our sim vehicles are close to 1 KG and the consistency helps switching between simulators.

**Comment by LorenzMeier** [comment]
I merged jMAVLib so this should pass now.

File: src/me/drton/jmavsim/Simulator.java (hunk 1 above)
Comment (kd0aij) [inline src/me/drton/jmavsim/Simulator.java]: at 0.8kg, the Iris hovers at about 50% throttle. Increasing the mass to 1kg requires over 60% throttle to hover. No one has replied to the issue I raised: https://github.com/PX4/jMAVSim/issues/32

**Comment by bkueng** [comment]
@kd0aij you need to update the git submodule link to point to the PX4 repo to fix the CI error.
https://github.com/PX4/jMAVSim/blob/master/.gitmodules#L3

**Comment by kd0aij** [comment]
@bkueng Thanks, I had forgotten that my fork of jMAVlib was from DrTon/jMAVlib. Unfortunately, PX4/jMAVSim master won't compile with PX4/jMAVlib but is OK with DrTon/jMAVlib. My testing has been done with kd0aij/jMAVSim and kd0aij/jMAVlib, so we need to find the divergence in PX4/jMAVlib.

Is a PX4 fork of jMAVlib really necessary?

**Comment by bkueng** [comment]
I would much prefer not to have different forks either. But I'd say it's the same as with FlightPlot: @DrTon told me he currently has no time to maintain it, so we started the PX4 fork, to which he agreed. The advantage of the PX4 repo is, that multiple people have write access (including DrTon), so updates don't get blocked/delayed too much.
If you want you can email him as well. Or if you have a better alternative, I'd like to hear it.

**Comment by kd0aij** [comment]
I agree that PX4/jMAVSim is necessary (and I have a number of 2 year old enhancements hanging around), but I was wondering about PX4/jMAVlib, since PX4/jMAVSim works with my fork of  DrTon/jMAVlib but not PX4/jMAVlib. 

I'll see if I can figure out the divergence between the kd0aij and PX4 jMAVlib forks.

**Comment by bkueng** [comment]
I was able to fix the compile error with:
```
--- a/src/me/drton/jmavsim/LogPlayerSensors.java
+++ b/src/me/drton/jmavsim/LogPlayerSensors.java
@@ -3,7 +3,7 @@ package me.drton.jmavsim;
 import me.drton.jmavlib.geo.LatLonAlt;
 import me.drton.jmavlib.log.FormatErrorException;
 import me.drton.jmavlib.log.LogReader;
-import me.drton.jmavlib.log.PX4LogReader;
+import me.drton.jmavlib.log.px4.PX4LogReader;

 import javax.vecmath.Vector3d;
 import java.io.EOFException;
```

**Comment by kd0aij** [comment]
How does that work? There's no log.px4 directory in PX4/jMAVlib

**Comment by bkueng** [comment]
yes there is: https://github.com/PX4/jMAVlib/tree/master/src/me/drton/jmavlib/log/px4

**Comment by kd0aij** [comment]
Had to edit .gitmodules directly to make it point at PX4. Just checking out master from PX4/jMAVlib was insufficient.

**Comment by bkueng** [comment]
This looks good now. Just to make sure this was your intention: you dropped the previous commits 'change Iris mass back to 0.8', 'change to HIL_STATE_QUATERNION', ... from this branch. Is that deliberate?

**Comment by kd0aij** [comment]
No, that was accidental, and I don't know how to recover them now.

**Comment by kd0aij** [comment]
maybe this one is right: https://github.com/PX4/jMAVSim/pull/34

**Pull request closed without merging**